### DISCUSSION TITLE
mqttv4 shouldn't die if it cannot connect

### DISCUSSION THREAD

**Discussion by olivluca** [roleoroleo/yi-hack-MStar#438] [Ideas]
I think that mqttv4 shouldn't die if it cannot connect, but should keep trying.
mqttv4.c does this [here](https://github.com/roleoroleo/mqttv4/blob/a0c3b2725b65dd124ffb6003c7afe124c2635601/src/mqttv4.c#L281)

```
    ret=mqtt_connect();
    if(ret!=0)
        exit(EXIT_FAILURE);
```

in turn mqtt_connect tries to connect MAX_RETRIES*10 and gives a failure if it cannot connect.
However, if the first connection succeeds, then, even if it's interrupted, it will be retried indefinitely

```
    while(1)
    {
        mqtt_check_connection();
        mqtt_loop();
        usleep(500*1000);
    }
```

I think that even the first connection should do the same. My use case is a camera connected to a cellular hotspot, so the connection can be spotty and if there's no connection established when mqttv4 starts it will eventually die.

Maybe it would be enough to just remove the `&& retries<=(MAX_RETRY*10)` conditions from both loops in mqtt_connect (though I'm not sure about the second one calling mosquitto_loop even if the connection failed).

**Comment by roleoroleo** [comment]
This is a good improvement.
Do you want to make a pr?
This is the repo:
https://github.com/roleoroleo/mqttv4

**Comment by olivluca** [comment]
Any hint on how to compile just mqttv4? My first attemps have been unsuccessful (the "MStar MSC3XX SDK.zip" I found has no "arm-hisiv300-linux-gcc" command, used in mqttv4 makefile).

**Reply by roleoroleo** [reply]
hisiv300 is the cross-compiler of the yi-hack-v4 version.
Check the scripts in this folder:
https://github.com/roleoroleo/yi-hack-MStar/tree/master/src/mqtt
Primarily config.mak

**Comment by olivluca** [comment]
I opened a pull request with a different approach
